modules/scheduler/gke-cluster/main.tf

```diff
@@ -30,6 +30,12 @@ locals {
   }]
 
   sa_email = var.service_account_email != null ? var.service_account_email : data.google_compute_default_service_account.default_sa.email
+
+  # additional VPCs enable multi networking 
+  derived_enable_multi_networking = length(var.additional_networks) > 0 ? true : coalesce(var.enable_multi_networking, false)
```

Comment:
I would reverse the logic on this. Since `enable_multi_networking` default is null, if it is set to either True or False then we know it is an explicit user action. We should not silently ignore this user action. 

Only if the variable is null then we should override with intuition based on additional network count.

Reply:
if additional_networks is passed, enable_multi_networking can be either null or explicitly True. Otherwise precondition blocks it.

Reply:
`derived_enable_multi_networking = coalesce(var.enable_multi_networking, length(var.additional_networks) > 0 ? true : false)`